lib/cli/options.js

```diff
@@ -104,24 +114,41 @@ const nargOpts = types.array
 const parse = (args = [], defaultValues = {}, ...configObjects) => {
   // save node-specific args for special handling.
   // 1. when these args have a "=" they should be considered to have values
-  // 2. if they don't, they just boolean flags
+  // 2. if they don't, they are just boolean flags
   // 3. to avoid explicitly defining the set of them, we tell yargs-parser they
   //    are ALL boolean flags.
   // 4. we can then reapply the values after yargs-parser is done.
-  const nodeArgs = (Array.isArray(args) ? args : args.split(' ')).reduce(
-    (acc, arg) => {
-      const pair = arg.split('=');
-      let flag = pair[0];
-      if (isNodeFlag(flag, false)) {
-        flag = flag.replace(/^--?/, '');
-        return arg.includes('=')
-          ? acc.concat([[flag, pair[1]]])
-          : acc.concat([[flag, true]]);
-      }
-      return acc;
-    },
-    []
-  );
+  const allArgs = Array.isArray(args) ? args : args.split(' ');
+  const nodeArgs = allArgs.reduce((acc, arg, index, allArgs) => {
+    const maybeFlag = allArgs[index - 1];
+
+    if (isNumeric(arg) && (!maybeFlag || !isMochaFlag(maybeFlag))) {
```

Comment:
[Question] Why `isNumeric(arg) &&`? Wouldn't we want to check this for other string flags too?

Unless I'm missing something, could this be trimmed down to?:

```suggestion
    if (!maybeFlag || !isMochaFlag(maybeFlag)) {
```

Reply:
Same here. I'll wait for your reply to [the previous comment](https://github.com/mochajs/mocha/pull/5263#discussion_r1869971177) before addressing this.

Reply:
I realized that this means that mocha does not currently accept filenames which only has numeric characters (example `mocha 404` is not accepted).